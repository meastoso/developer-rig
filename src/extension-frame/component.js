import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';
const { ExtensionPlatform } = window['extension-coordinator'];

export const EXT_FRAME_WRAPPER_CLASS = 'extension-frame-wrapper';

export class ExtensionFrame extends Component {
  constructor() {
    super(...arguments);

    this._boundOnFrameDoubleClick = this._onFrameDoubleClick.bind(this);
    this._boundOnIdentityLinked = this._onIdentityLinked.bind(this);
    this._boundOnModalRequested = this._onModalRequested.bind(this);
    this._boundIframeHostRefHandler = this._iframeHostRefHandler.bind(this);

  }

  componentDidMount() {
    const { ExtensionFrame } = window['extension-coordinator'];
    this.extensionFrame = new ExtensionFrame({
      anchor: this.props.type,
      channelId: this.props.extension.channelId,
      loginId: null,
      dobbin: {
        trackEvent: (eventName, properties, services) => { },
      },
      extension: this.props.extension,
      iframeClassName: this.props.iframeClassName,
      mode: this.props.mode,
      platform: ExtensionPlatform.Web,
      parentElement: this.parentElement,
      trackingProperties: {},
    });

    if (this.parentElement) {
      this.parentElement.addEventListener('dblclick', this._boundOnFrameDoubleClick);
    }
    this.extensionFrame.on('identityLinked', this._boundOnIdentityLinked);
    this.extensionFrame.on('requestModal', this._boundOnModalRequested);
  }

  render() {
    return (
      <div
        ref={this._boundIframeHostRefHandler}
        className={EXT_FRAME_WRAPPER_CLASS}
      />
    );
  }

  _onIdentityLinked(isLinked) {
    const { extension, onIdentityLinked } = this.props;
    if (onIdentityLinked) {
      onIdentityLinked(extension.id, isLinked);
    }
  }

  _onFrameDoubleClick(evt) {
    evt.preventDefault();
  }

  _onModalRequested(confirmationRequest){}

  _iframeHostRefHandler(element) {
    this.parentElement = element;
  }
}

ExtensionFrame.propTypes = {
  iframeClassName: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
};
