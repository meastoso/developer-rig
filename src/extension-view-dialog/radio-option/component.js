import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class RadioOption extends Component {
  render() {
    return (
      <div>
        <label className="option-label">
            <input type="radio" name={this.props.name} value={this.props.value} onChange={this.props.onChange} defaultChecked={this.props.checked}/>
            {this.props.value}
        </label>
      </div>
    )
  }
}

RadioOption.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool
}
