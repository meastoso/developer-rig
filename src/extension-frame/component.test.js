import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ExtensionForTest } from '../tests/constants/extension';
import { ExtensionFrame } from './component';

const { ExtensionViewType, ExtensionAnchor, ExtensionMode } = window['extension-coordinator'];

describe('<ExtensionFrame />', () => {
  const setupShallow = setupShallowTest(ExtensionFrame, () => ({
    iframeClassName: 'rig-frame frameid-0',
    extension: ExtensionForTest,
    type: ExtensionAnchor.Panel,
    mode: ExtensionMode.Viewer,
  }));

  it('prevents the default when double clicked', () => {
    const mockEvent = {};
    mockEvent.preventDefault = jest.fn();
    const { wrapper } = setupShallow();
    wrapper.instance()._onFrameDoubleClick(mockEvent);
    expect(mockEvent.preventDefault).toHaveBeenCalled();
  });

  it('onIdentityLink prop is called', () => {
    const { wrapper } = setupShallow({
      onIdentityLinked: jest.fn()
    });
    wrapper.instance()._onIdentityLinked();
    expect(wrapper.instance().props.onIdentityLinked).toHaveBeenCalled();
  });

  it('parentElement is set when _iframeHostRefHandler is called', () => {
    const { wrapper } = setupShallow();
    const mockParent = 'test';
    wrapper.instance()._iframeHostRefHandler(mockParent);
    expect(wrapper.instance().parentElement).toBe(mockParent);
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.LiveConfig,
        mode: ExtensionMode.Dashboard,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.Config,
        mode: ExtensionMode.Config,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in panel mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Panel,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Overlay,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
