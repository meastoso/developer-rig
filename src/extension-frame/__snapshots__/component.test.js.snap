// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionFrame /> when in config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    iframeClassName="rig-frame frameid-0"
    mode="config"
    type="config"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "extension-frame-wrapper",
    },
    "ref": [Function],
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "extension-frame-wrapper",
      },
      "ref": [Function],
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in live config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    iframeClassName="rig-frame frameid-0"
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "extension-frame-wrapper",
    },
    "ref": [Function],
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "extension-frame-wrapper",
      },
      "ref": [Function],
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in panel mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    iframeClassName="rig-frame frameid-0"
    mode="viewer"
    type="panel"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "extension-frame-wrapper",
    },
    "ref": [Function],
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "extension-frame-wrapper",
      },
      "ref": [Function],
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in video overlay mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    iframeClassName="rig-frame frameid-0"
    mode="viewer"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "extension-frame-wrapper",
    },
    "ref": [Function],
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "extension-frame-wrapper",
      },
      "ref": [Function],
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
